src/backend/updater/updater.ts

```diff
@@ -91,16 +109,38 @@ autoUpdater.on('update-downloaded', async () => {
   if (response === 1) {
     return autoUpdater.quitAndInstall()
   }
-  hasUpdated = true
+  logInfo('User chose not to update the app for now.', LogPrefix.AutoUpdater)
 })
 
 autoUpdater.on('error', async (error) => {
-  if (!isOnline()) {
+  isAppUpdating = false
+  const isNewVersion = newVersion !== app.getVersion()
+
+  // To avoid false positives, we should not show the error dialog if the app has already updated successfully
+  if (hasUpdated || !isNewVersion) {
```

Comment:
I am afraid this won't still get 100% of false positives since the API call can still fail when trying to reach Github due to internet issues, for instance, that is why I am adding a new event above to track the downloading progress. So then we can have a funel to get only actuall download and install errors.